JS files are not ignored properly

### Describe the bug

ESLint still says
```
  0:0  error  Parsing error: ESLint was configured to run on `<tsconfigRootDir>/eslint.config.js`
              using `parserOptions.project`: <tsconfigRootDir>/tsconfig.json
```

JS files will error if they have 'project' field no matter if there are ts-aware rules or not as I understand

### Reproduction

https://github.com/Dimava/antfu-eslint-config/tree/repro/1 (a clean project basically)

### System Info

```Shell
version 2.6.1 (latest)
```


### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/antfu/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/antfu/contribute).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

--- a/src/configs/typescript.ts
+++ b/src/configs/typescript.ts
@@ -19,6 +19,10 @@ export async function typescript(
   ]
 
   const filesTypeAware = options.filesTypeAware ?? [GLOB_TS, GLOB_TSX]
+  const tsconfigPath = options?.tsconfigPath
+    ? toArray(options.tsconfigPath)
+    : undefined
+  const isTypeAware = !!tsconfigPath
 
   const typeAwareRules: FlatConfigItem['rules'] = {
     'dot-notation': 'off',
@@ -42,10 +46,6 @@ export async function typescript(
     'ts/unbound-method': 'error',
   }
 
-  const tsconfigPath = options?.tsconfigPath
-    ? toArray(options.tsconfigPath)
-    : undefined
-
   const [
     pluginTs,
     parserTs,
@@ -53,6 +53,28 @@ export async function typescript(
     interopDefault(import('@typescript-eslint/eslint-plugin')),
     interopDefault(import('@typescript-eslint/parser')),
   ] as const)
+
+  function makeParser(typeAware: boolean, files: string[], ignores?: string[]): FlatConfigItem {
+    return {
+      files,
+      ...ignores ? { ignores } : {},
+      languageOptions: {
+        parser: parserTs,
+        parserOptions: {
+          extraFileExtensions: componentExts.map(ext => `.${ext}`),
+          sourceType: 'module',
+          ...typeAware
+            ? {
+                project: tsconfigPath,
+                tsconfigRootDir: process.cwd(),
+              }
+            : {},
+          ...parserOptions as any,
+        },
+      },
+      name: `antfu:typescript:${typeAware ? 'type-aware-parser' : 'parser'}`,
+    }
+  }
 
   return [
     {
@@ -63,22 +85,15 @@ export async function typescript(
         ts: pluginTs as any,
       },
     },
+    // assign type-aware parser for type-aware files and type-unaware parser for the rest
+    ...isTypeAware
+      ? [
+          makeParser(true, filesTypeAware),
+          makeParser(false, files, filesTypeAware),
+        ]
+      : [makeParser(false, files)],
     {
       files,
-      languageOptions: {
-        parser: parserTs,
-        parserOptions: {
-          extraFileExtensions: componentExts.map(ext => `.${ext}`),
-          sourceType: 'module',
-          ...tsconfigPath
-            ? {
-                project: tsconfigPath,
-                tsconfigRootDir: process.cwd(),
-              }
-            : {},
-          ...parserOptions as any,
-        },
-      },
       name: 'antfu:typescript:rules',
       rules: {
         ...renameRules(

